sigopt_sklearn/search.py

```diff
@@ -150,6 +150,10 @@ def __init__(self, estimator, param_domains, n_iter=10, scoring=None,
         self.opt_timeout = opt_timeout
         self.verbose = verbose
 
+        # Stores the mappings between categorical strings to Python values. The keys correspond to parameter names and
+        # values correspond to the string-to-value mappings themselves.
+        self.categorical_mappings_ = {}
```

Comment:
I'm not opposed to the idea, but I'm just not entirely sure how this would look in Python. I think that this would require subclassing `dict`, which may turn out to be a bit verbose. One way to think about the current implementation is as `_convert_nonstring_categoricals` being the general mapping that you describe, and `categorical_mappings_` as providing some internal state required by `_convert_nonstring_categoricals`.

What do you think?